modules/gdnative/gdnative.cpp

```diff
@@ -858,12 +868,56 @@ bool GDNativeInstance::get(const StringName &p_name, Variant &r_ret) const {
 		r_ret = *(Variant *)&value;
 		return true;
 	}
+
+	Map<StringName, GDNativeScriptData::Method>::Element *E = script->script_data->methods.find("_get");
+	if (E) {
+		Variant name = p_name;
+		const Variant *args[1] = { &name };
+
+		godot_variant result = E->get().method.method((godot_object *)owner, E->get().method.method_data, userdata, 1, (godot_variant**) args);
+		r_ret = *(Variant *)&result;
+		return true;
+	}
+
 	return false;
 }
 
 void GDNativeInstance::get_property_list(List<PropertyInfo> *p_properties) const {
 	script->get_script_property_list(p_properties);
 	// TODO: dynamic properties
+
+	Map<StringName, GDNativeScriptData::Method>::Element *E =  script->script_data->methods.find("_get_property_list");
+	if (E) {
+		godot_variant result = E->get().method.method((godot_object *)owner, E->get().method.method_data, userdata, 0, NULL);
+		Variant ret = *(Variant *)&result;
```

Comment:
That "weird" C-style cast is used like everywhere in the engine (apart from some parts of the compile time generated PTRCALL converters and some Reference casting code which also use `reinterpret_cast`). The whole engine doesn't really use C++ style casts, and this is done so many times here that using a C++ style cast would just make it all less readable. 

It's not in a different function because it's more to write and doesn't make it any more readable either. Using a function suggest an action or calculation, both of which don't happen here. This is just the conversion between C land and C++ land and it's short and sweet IMO.

We're shovelling data from C++ to C and on the other end to C++, D, Rust or Go. I think const correctness in the C++ part is our smallest possible point of error here.